music21/metadata/__init__.py

```diff
@@ -39,12 +39,95 @@
+    are md['dcterms:alternative'] and md['alternativeTitle']. You can have more than
+    one of any such item (not just contributors).  And you can also have metadata
+    items with custom names.
+
+    If you are only interested in getting/setting a single string for a metadata item
+    (even if there is more than one item of that name), you can use an attribute-style
+    get operation (e.g. t = md.title).  If you want to see a list of metadata items in
+    their appropriate value type, you can us a dictionary-style get operation (e.g.
+    titles = md['title']).  If you set an item or list of items (via attribute-style
+    or dictionary-style), any existing items of that name are deleted.  If you want
+    to add an item or list of items without deleting existing items, you can use the
+    md.add() API.  See the examples below:
+
+    Set a title (overwrites any existing titles):
+
+    >>> md = metadata.Metadata()
+    >>> md.title = 'A Title'
+    >>> md.title
+    'A Title'
+    >>> md['title']
+    (<music21.metadata.primitives.Text A Title>,)
+
+    Set two titles (overwrites any existing titles):
+
+    >>> md['title'] = ['The Title', 'A Second Title']
+    >>> md['title']
+    (<music21.metadata.primitives.Text The Title>,
+    <music21.metadata.primitives.Text A Second Title>)
+    >>> md.title
+    'The Title, A Second Title'
+
+    Add a third title (leaves any existing titles in place):
+
+    >>> md.add('title', 'Third Title, A')
+    >>> md['title']
+    (<music21.metadata.primitives.Text The Title>,
+    <music21.metadata.primitives.Text A Second Title>,
+    <music21.metadata.primitives.Text Third Title, A>)
+    >>> md.title
+    'The Title, A Second Title, A Third Title'
+
+    You can also set/add/get free-form custom metadata items:
+
+    >>> md.setCustom('modification description', 'added missing sharp in measure 27')
+    >>> md.getCustom('modification description')
+    (<music21.metadata.primitives.Text added missing sharp in measure 27>,)
+    >>> md.addCustom('modification description', 'deleted redundant natural in measure 28')
+    >>> md.getCustom('modification description')
+    (<music21.metadata.primitives.Text added missing sharp in measure 27>,
+    <music21.metadata.primitives.Text deleted redundant natural in measure 28>)
+
+    Primitives: primitives.Text has been updated to add whether or not the text has
+    been translated, as well as a specified encoding scheme (a.k.a. what standard
+    should be used to parse this string).
+
+    Metadata does not explicitly support client-specified namespaces, but by using
+    getCustom/addCustom/setCustom, clients can set anything they want. Clients could
+    use this to set (say) 'myCoolScoreFileFormat:XXX' metadata that doesn't map to
+    any standard metadata property, and a writer of myCoolScoreFileFormat files that
+    understood 'myCoolScoreFileFormat' metadata could then write it to one of
```

Comment:
is there an example of an actual obsolete format that we never intend to support that might be a better example than "myCoolScoreFileFormat:XXX"  ?    

```
For instance, to embed the old SoundTracker .MOD format's sample name, you could use:
`soundtracker:SampleName`
```

etc.  

Any time we give users an example of something cool they might do with a feature, they're more likely to put the system to its full potential.

(Okay, now I want to write a music21 parser for .MOD files -- the best integration of symbolic and audio music formats ever invented: http://www.aes.id.au/modformat.html )

Reply:
I'll switch to your .MOD example.  I had Humdrum as my example here, but that's not quite as straightforward, since we are using parts of the Humdrum namespace in the non-custom APIs.

Reply:
Done.